src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = new Object[INITIAL_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        validateIndex(index, size + 1);
```

Comment:
All magic numbers, strings, symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)

Reply:
The one here is not a "magic number" but the extent to size of the array to tell that we should check the index bound including next element,

Reply:
Following your logic...
Should I change these lines? i.e. replace 1 to a constant in "index + 1"?

System.arraycopy(elements, index, elements, index + 1, size - index);
System.arraycopy(elements, index + 1, elements, index, size - index);

please advise. thanks!